python/ipywidgets/ipywidgets/widgets/widget.py

```diff
@@ -272,22 +283,46 @@ def register(widget):
     return widget
 
 
+class _staticproperty(object):
+    def __init__(self, fget):
+        self.fget = fget
+
+    def __get__(self, owner_self, owner_cls):
+        assert owner_self is None
+        return self.fget()
+
+
+
 class Widget(LoggingHasTraits):
     #-------------------------------------------------------------------------
     # Class attributes
     #-------------------------------------------------------------------------
     _widget_construction_callback = None
     _control_comm = None
 
-    # _active_widgets is a dictionary of all active widget objects
-    _active_widgets = {}
+    @_staticproperty
+    def widgets():
+        warnings.warn("Widget.widgets is deprecated, use ipywidgets.widgets.widget._instances", DeprecationWarning)
+        return _instances
+
+    @_staticproperty
+    def _active_widgets():
+        warnings.warn("Widget._active_widgets is deprecated, use ipywidgets.widgets.widget._instances", DeprecationWarning)
```

Comment:
Since _instances is now a module private variable, can we not advertise it as if it was a supported public API?

Reply:
What kind of wording did you have in mind?
I'm also not 100% sure we should consider it 'private', more like 'protected'/'expert mode'. I think people can use it, but with the `_` we give a warning saying "this is internal stuff, we don't document the behavior, read the code", and "this shouldn't be normal user code, but it's there if you really need to."

FYI: I'm gonna depend on this dict a lot for the coming period :)

Reply:
> this is internal stuff, we don't document the behavior, read the code",

That's exactly what I mean by "private". And that if the behavior changes, it's up to you to fix your code. I just think we shouldn't advertise it in the deprecation message as if it was fully supported for people to use as a public API.

Of course, this *is* python, which means that anyone can get into anything...

Reply:
> What kind of wording did you have in mind?

```suggestion
        warnings.warn("Widget._active_widgets is deprecated", DeprecationWarning)
```